packages/styling/src/styles/theme.ts

```diff
@@ -43,6 +44,32 @@ export function getTheme(): ITheme {
 }
 
 /**
+ * Registers a callback that gets called whenever the theme changes.
+ * This should only be used when the component cannot automatically get theme changes through its state.
+ * This will not register duplicate callbacks.
+ */
+export function registerOnThemeChangeCallback(callback: (theme: ITheme) => void): void {
```

Comment:
Can we add some simple unit tests here?

Reply:
A test that makes sure loading a theme calls the callback would also be nice. Jest.fn() for mocking is great for that!

Reply:
added tests, could you take another look please?

Reply:
Looks good!